### PR TITLE
7pm-2-ViewSearchesAsAdmin

### PR DESCRIPTION
**Pull request by bowenduanmu**
Maintain the page to show table for searches data in SearchHistory page and modify the SearchHistory controller for admin user.

### CODE REVIEW

File: javascript/src/main/App.js [hunk 1]
```diff
@@ -46,10 +46,10 @@ function App() {
           <AuthorizedRoute path="/admin/searchedTerms" exact component={SearchedTerms} authorizedRoles={["admin"]} />
           <AuthorizedRoute path="/member/channels" exact component={ChannelList} authorizedRoles={["admin","member"]} />
           <AuthorizedRoute path="/member/search" exact component={Search} authorizedRoles={["admin","member"]} />
+          <AuthorizedRoute path="/member/searchhistory" exact component={SearchHistory} authorizedRoles={["admin","member"]} />
           <AuthorizedRoute path="/member/messages/search" component={SearchResults} authorizedRoles={["admin","member"]} />
           <AuthorizedRoute path="/member/analyzemessages/reactions" exact component={AnalyzeReactions} authorizedRoles={["admin", "member"]} />
           <AuthorizedRoute path="/member/listViewChannels/:channel" component={ChannelPageList} authorizedRoles={["admin","member"]} />
-          <AuthorizedRoute path="/member/channels/:channel/links" component={ChannelPageLinks} authorizedRoles={["admin","member"]} />
           <AuthorizedRoute path="/member/channels/:channel" component={ChannelPageScrollable} authorizedRoles={["admin","member"]} />
```
Comment (pconrad) [inline javascript/src/main/App.js:53]: Still don't know why you are removing this line.     See Reviews on #305 and #307

File: src/main/java/edu/ucsb/mapache/controllers/SearchHistoryController.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        return new ResponseEntity<String>(body, HttpStatus.UNAUTHORIZED);
+    }
+
+    @GetMapping("/allusersearches")
+    public ResponseEntity<String> getSearches(@RequestHeader("Authorization") String authorization)
+            throws JsonProcessingException {
+        if (!authControllerAdvice.getIsMember(authorization))
+            return getUnauthorizedResponse("member");
+        
+        String body=null;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/mapache/controllers/SearchHistoryController.java:66]: Lines 66, 68-69 and 71 should be covered by tests.

File: src/test/java/edu/ucsb/mapache/controllers/SearchHistoryControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    // public ResponseEntity<String> getSearches(@RequestHeader("Authorization")
+    // String authorization)
+
+    // I wrote 'case 1' because you'll probably need a few cases, e.g.
+    // for not logged in, logged in as admin, logged in as member, etc.
+    // depending on what the code is checking for
+  }
+
+  @Test
+  public void test_getSearches_case2() throws Exception {
```
Comment (pconrad) [inline src/test/java/edu/ucsb/mapache/controllers/SearchHistoryControllerTests.java:129]: I am really confused why there are three different PRs that seem to be covering the same territory, and why there is commented out code still in two of them. 

Should these be closed or marked as drafts?

**Review by pconrad — CHANGES_REQUESTED** [review]
It is really confusing that there are three open PRs, #305, #306, and #307 all seeming to be covering the same territory, in various stages of completion. 

Going through each of these separately wastes the time of the team and the staff, and time is running out.  

Please figure out which of these is the "real" one, and close the other two.  Or if they are really are focussed on different issues, make that more clear.

And the ones that aren't ready for review yet (e.g. have large sections of commented out code still to be written, mark them as drafts. )

**Comment by bowenduanmu** [comment]
This PR is included in another PR.

**Pull request closed without merging**